### PR TITLE
detect/sip: add sticky buffers to match headers v2

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6374

Describe changes:
- Match headers in compact form
- Add doc
- SV: add test for compact form

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1787

### CODE REVIEW

File: doc/userguide/upgrade.rst [hunk 1]
```diff
@@ -52,6 +52,14 @@ Major changes
   for the ``stats`` event.
 - Stats counters that are 0 can now be hidden from EVE logs. Default behavior
   still logs those (see :ref:`EVE Output - Stats <eve-json-output-stats>` for configuration setting).
+- The following sticky buffers for matching SIP headers have been implemented:
+    - sip.via
```
Comment (catenacyber) [inline doc/userguide/upgrade.rst:56]: It looks like these miss in doc/userguide/rules/sip-keywords.rst ;-)

File: rust/src/sip/parser.rs [hunk 2]
```diff
@@ -50,6 +50,7 @@ pub struct Response {
     pub version: String,
     pub code: String,
     pub reason: String,
+    pub headers: HashMap<String, String>,
```
Comment (catenacyber) [inline rust/src/sip/parser.rs:53]: What happens if I have multiple `via` headers ?

**Review by catenacyber — CHANGES_REQUESTED** [review]
Thanks Giuseppe, will need at least a rebase, and doc update for these new keywords

**Comment by catenacyber** [comment]
Also for your info, you can take a look at https://github.com/OISF/suricata/pull/10966

File: rust/src/sip/parser.rs (hunk 2 above)
Comment (glongo) [reply rust/src/sip/parser.rs:53]: The value is updated and the old one is returned.

**Comment by glongo** [comment]
Replaced with https://github.com/OISF/suricata/pull/11004

**Pull request closed without merging**